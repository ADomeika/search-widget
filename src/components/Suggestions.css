.suggestions {
  border: 1px solid #262626;
  border-radius: 2px;
  position: absolute;
  top: 100%;
  width: 100%;
}

.suggestions__list {
  list-style: none;
  margin: 0;
  padding: 0;
}

.suggestions__list-item {
  background-color: #fff;
  cursor: pointer;
}

.suggestions__list-item:hover {
  background-color: #e7f4fe;
}

.suggestion {
  align-items: flex-start;
  display: flex;
  flex-wrap: nowrap;
  padding: 8px;
}

.suggestion__no-results {
  background-color: #fff;
  font-size: 14px;
  padding: 8px;
}

.suggestion__badge {
  border: 1px solid transparent;
  border-radius: 2px;
  color: #fff;
  font-size: 12px;
  font-weight: 500;
  line-height: 1rem;
  min-width: 5rem;
  padding: 2px 4px;
  text-align: center;
}

.suggestion__badge--airport {
  background-color: #961412;
}

.suggestion__badge--city {
  background-color: #0a63b0;
}

.suggestion__badge--station {
  background-color: #494949;
}

.suggestion__badge--district {
  background-color: #017c44;
}

.suggestion__badge--region {
  color: #222;
  background-color: #f1c74c;
}

.suggestion__info-holder {
  padding-left: 8px;
}

.suggestion__title {
  font-size: 14px;
  font-weight: 500;
  line-height: 20px;
}

.suggestion__subtext {
  font-size: 12px;
  font-weight: 500;
  line-height: 18px;
}

.hr {
  border: 0;
  border-top: 1px solid #e7e7e7;
  margin: 0;
}
