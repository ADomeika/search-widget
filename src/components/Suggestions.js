import React from 'react';
import './Suggestions.css';

const Suggestions = ({ suggestions }) => {
  const getPlaceType = (typeChar) => {
    switch (typeChar) {
      case 'A':
        return 'Airport';
      case 'C':
        return 'City';
      case 'T':
        return 'Station';
      case 'D':
        return 'District';
      case 'P':
        return 'Region';
      default:
        return 'Unknown';
    }
  };

  return (
    <div className='suggestions'>
      <ul className='suggestions__list'>
        {suggestions.map((suggestion) => {
          if (suggestion.name === 'No results found') {
            return (
              <div className='suggestion__no-results' key='no-results'>No results found</div>
            );
          }
          return (
            <li key={suggestion.placeKey} className='suggestions__list-item'>
              <div className='suggestion'>
                <span
                  className={`suggestion__badge suggestion__badge--${getPlaceType(suggestion.placeType).toLowerCase()}`}
                >
                  {getPlaceType(suggestion.placeType)}
                </span>
                <div className='suggestion__info-holder'>
                  <div className='suggestion__title'>
                    {suggestion.name} {suggestion.iata ? `(${suggestion.iata})` : ''}
                  </div>
                  <div className='suggestion__subtext'>
                    {suggestion.city ? `${suggestion.city}, ` : ''}
                    {suggestion.region ? `${suggestion.region}, ` : ''}
                    {suggestion.country}
                  </div>
                </div>
              </div>
              <hr className='hr' />
            </li>
          );
        })}
      </ul>
    </div>
  );
};

export default Suggestions;
