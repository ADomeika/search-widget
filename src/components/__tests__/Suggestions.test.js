import React from 'react';
import { shallow } from 'enzyme';

import Suggestions from '../Suggestions';

const mockSuggestions = [
  {
    name: 'Manchester Airport',
    placeKey: '1',
    placeType: 'A',
    iata: 'MAN',
    city: 'Manchester',
    region: 'Greater Manchester',
    country: 'United Kingdom'
  },
];

describe('<Suggestions />', () => {
  it('should render', () => {
    expect(shallow(<Suggestions suggestions={mockSuggestions} />)).toMatchSnapshot();
  });

  it('should render no results found', () => {
    const mockNoSuggestions = [{ name: 'No results found' }];
    const wrapper = shallow(<Suggestions suggestions={mockNoSuggestions} />);

    expect(wrapper.find('.suggestion__no-results').exists()).toBeTruthy();
  });
});
