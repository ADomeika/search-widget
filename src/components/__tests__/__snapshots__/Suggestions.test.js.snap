// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Suggestions /> should render 1`] = `
<div
  className="suggestions"
>
  <ul
    className="suggestions__list"
  >
    <li
      className="suggestions__list-item"
      key="1"
    >
      <div
        className="suggestion"
      >
        <span
          className="suggestion__badge suggestion__badge--airport"
        >
          Airport
        </span>
        <div
          className="suggestion__info-holder"
        >
          <div
            className="suggestion__title"
          >
            Manchester Airport
             
            (MAN)
          </div>
          <div
            className="suggestion__subtext"
          >
            Manchester, 
            Greater Manchester, 
            United Kingdom
          </div>
        </div>
      </div>
      <hr
        className="hr"
      />
    </li>
  </ul>
</div>
`;
